src/uct/sm/scopy/knem/knem_md.c

```diff
@@ -201,8 +201,9 @@ ucs_status_t uct_knem_md_query(uct_md_h uct_md, uct_md_attr_v2_t *md_attr)
     md_attr->reg_mem_types          = 0;
     md_attr->reg_nonblock_mem_types = 0;
     if (uct_knem_md_check_mem_reg(uct_md)) {
-        md_attr->flags         |= UCT_MD_FLAG_REG;
-        md_attr->reg_mem_types |= UCS_BIT(UCS_MEMORY_TYPE_HOST);
+        md_attr->flags                  |= UCT_MD_FLAG_REG;
+        md_attr->reg_mem_types          |= UCS_BIT(UCS_MEMORY_TYPE_HOST);
+        md_attr->reg_nonblock_mem_types |= UCS_BIT(UCS_MEMORY_TYPE_HOST);
```

Comment:
How is this change related to IB ODP?

Reply:
knem rehistration is pinning the memory, so it is not really supporting HOST memory as "nonblock"
UCP logic should not register with knem when nonblock registration is required